src/runtime/components/forms/Rating.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div :class="ui.wrapper">
+    <UIcon
+      v-for="star of [...Array(max).keys()].slice().reverse()"
+      :key="star"
+      :name="icon"
+      :class="[
+        ratingClass,
+        (star + 1 <= modelValue ? ui.active : ui.inactive).replaceAll(
+          '{color}',
+          color
+        ),
+        readOnly ? 'cursor-auto hover:text-gray-300 hover:dark:text-gray-700 peer-hover:text-gray-300 peer-hover:dark:text-gray-700' : 'cursor-pointer',
+        'order-{order}'.replaceAll('{order}', star.toString()),
+      ]"
+      @click="setRating(star + 1)"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { computed } from 'vue'
+import { defu } from 'defu'
+import UIcon from '../elements/Icon.vue'
+import { classNames } from '../../utils'
+import { useFormEvents } from '../../composables/useFormEvents'
+import { useAppConfig } from '#imports'
+
+// TODO: Remove
+// @ts-expect-error
+import appConfig from '#build/app.config'
+
+export default defineComponent({
+  components: {
+    UIcon
+  },
+  props: {
+    modelValue: {
+      type: Number,
+      default: 0
+    },
+    icon: {
+      type: String,
+      default: () => appConfig.ui.rating.default.icon
+    },
+    max: {
+      type: Number,
+      default: () => appConfig.ui.rating.default.max
```

Comment:
I believe it would be good to add `max` value here. and add `validate` for the max value 
```vue
  max: {
    type: Number,
    default: 5,
    validate (value) {
      return value >= 1 && value < Number.MAX_VALUE
    }
  },
```